Tavisca.Bootcamp.LanguageBasics.Exercise3/Program.cs

```diff
@@ -28,20 +29,232 @@ static void Main(string[] args)
+            int[] calories = new int[protein.Length];
+            int[] outputDiets = new int[dietPlans.Length];
+            int[] dietsWithMinimumProtein = frequency.FrequencyCalculator( protein,protein.Min());
+            int[] dietsWithMaximumProtein = frequency.FrequencyCalculator(protein, protein.Max());
+            int[] dietsWithMinimumCarbs = frequency.FrequencyCalculator(carbs, carbs.Min());
+            int[] dietsWithMaximumCarbs = frequency.FrequencyCalculator(carbs, carbs.Max());
+            int[] dietsWithMinimumFat = frequency.FrequencyCalculator(fat, fat.Min());
+            int[] dietsWithMaximumFat = frequency.FrequencyCalculator(fat, fat.Max());
+            //now each of above array will contain indices of diets eligible. will contain more than one in case of ties.
+            //following will calculate protein ammount in each diet.
+            for (int i = 0; i < protein.Length; i++)
+            {
+                calories[i] = 9 * fat[i] + 5 * (protein[i] + carbs[i]);
+            }
+            int[] dietsWithMinimumCalories = frequency.FrequencyCalculator(calories, calories.Min());
+            int[] dietsWithMaximumCalories = frequency.FrequencyCalculator(calories, calories.Max());
+            /*Now the logic is simple we will first check first character of given string and will check whether length of it's 
+            array is grater than 1 or not if not then the first element is our answer if no ww will go for next char if next char
+            has index which was present in earlier array then that index is our answer if not then go for next char or first element from
+            previous array is our answer.*/
+            for (int i = 0; i < dietPlans.Length; i++)
+            {
+                char[] diet = dietPlans[i].ToCharArray();
+                int j = 0;
+                IList<int> dietsTillNow = new List<int>();
+                bool flag = false;
+                while (j < diet.Length)
+                {
+
+                    if (diet[j] == 'p')
+                    {
+                        if (dietsWithMinimumProtein.Length > 1)
+                        {
+                            foreach (int k in dietsWithMinimumProtein) dietsTillNow.Add(k);
+                            flag = true;
+                            j++;
+                        }
+                        else
+                        {
+                            if (flag && !dietsTillNow.Contains(dietsWithMinimumProtein[0]))
+                            {
+                                outputDiets[i] = dietsTillNow[0];
+                                j++;
+                            }
+                            else
+                            {
+                                outputDiets[i] = dietsWithMinimumProtein[0];
+                                break;
+                            }
+                        }
+                    }
+                    else if (diet[j] == 'P')
+                    {

```

Comment:
Look for the way where we can reduce the number of if /else statements.

Read these

https://stackoverflow.com/questions/46800290/reduce-the-amount-of-if-else-statements-and-simpler-code

https://softwareengineering.stackexchange.com/questions/206816/clarification-of-avoid-if-else-advice

https://kodify.net/csharp/if-else/if-else-to-avoid/

https://codereview.stackexchange.com/questions/33463/optimizing-multiple-if-else-from-static-class-with-linq

Reply:
Sir I have used dictionary now please check the code now

Reply:
@tavisca-vtripathi